lib/appium-driver.ts

```diff
@@ -524,42 +564,53 @@ export class AppiumDriver {
         return await this.driver.getSessionId();
     }
 
-    public async compareElement(element: UIElement, imageName: string, tolerance: number = 0.01, timeOutSeconds: number = 3, toleranceType?: ImageOptions) {
+    public async compareElement(element: UIElement, imageName?: string, tolerance: number = 0, timeOutSeconds: number = 3, toleranceType: ImageOptions = ImageOptions.percent) {
         return await this.compareRectangle(await element.getActualRectangle(), imageName, timeOutSeconds, tolerance, toleranceType);
     }
 
-    public async compareRectangle(rect: IRectangle, imageName: string, timeOutSeconds: number = 3, tolerance: number = 0.01, toleranceType?: ImageOptions) {
-        return await this.compare(imageName, timeOutSeconds, tolerance, rect, toleranceType);
+    public async compareRectangle(rect: IRectangle, imageName?: string, timeOutSeconds: number = 3, tolerance: number = 0, toleranceType: ImageOptions = ImageOptions.percent) {
+        imageName = imageName || this.imageHelper.testName;
+        const options = this.imageHelper.extendOptions({
+            imageName: imageName,
+            timeOutSeconds: timeOutSeconds,
+            tolerance: tolerance,
+            cropRectangle: rect,
+            toleranceType: toleranceType,
+            keepOriginalImageName: true,
+            keepOriginalImageSize: false
+        });
+        return await this.imageHelper.compare(options);
     }
 
-    public async compareScreen(imageName: string, timeOutSeconds: number = 3, tolerance: number = 0.01, toleranceType?: ImageOptions) {
-        return await this.compare(imageName, timeOutSeconds, tolerance, undefined, toleranceType);
+    public async compareScreen(imageName?: string, timeOutSeconds: number = 3, tolerance: number = 0, toleranceType: ImageOptions = ImageOptions.percent) {
```

Comment:
Can we preserve toleranceType and tolerance as optional since they both have default value